### PR TITLE
frontend/feat: 1410 Updated homescreen with older design

### PR DESCRIPTION
**Pull request by aryan-13**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

Updated homescreen with older design
Firma - [LINK](https://www.figma.com/file/4zT4QhImxqphGf56DpAjcV/Beckn-Release-Version?type=design&node-id=95302-28611&mode=design&t=27ZerwJnyqVh04GN-4)

Screenshot 2023-12-15 at 5 37 41 PM

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 1]
```diff
@@ -305,9 +323,19 @@ view push state =
...
+                  --   if isHomeScreenView state then homeScreenView push state else emptyTextView state
+                  -- , if isHomeScreenView state then emptyTextView state else mapView push state
+                  --   ]
+                   linearLayout
+                     [ height if any (_ == state.props.currentStage) [RideAccepted, RideStarted, ChatWithDriver] && os /= "IOS" 
+                                then (V (((screenHeight unit)/ 15)*10)) 
+                                else MATCH_PARENT
+                     , width MATCH_PARENT
+                     , accessibility DISABLE_DESCENDANT
+                     , id (getNewIDWithTag "CustomerHomeScreenMap")
```
Comment (PraveenGongada) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:335]: use `$`

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 2]
```diff
@@ -305,9 +323,19 @@ view push state =
...
+                     , visibility if state.props.isSrcServiceable then VISIBLE else GONE
```
Comment (PraveenGongada) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:336]: use `boolToVisibility` from Mobility.Prelude

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 3]
```diff
@@ -638,12 +668,14 @@ shareAppPopUp push state =
...
-  [ height WRAP_CONTENT
-  , width MATCH_PARENT
-  , alignParentBottom "true,-1"
-  , orientation VERTICAL
-  ][ if (state.props.currentStage == HomeScreen && (not state.props.rideRequestFlow) && (not state.props.showlocUnserviceablePopUp)) then buttonLayout state push else emptyTextView state]
+  PrestoAnim.animationSet (buttonLayoutAnimation state) $
+      linearLayout
+      [ height WRAP_CONTENT
+      , width MATCH_PARENT
+      , id (getNewIDWithTag "buttonLayout")
```
Comment (PraveenGongada) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:675]: use `$`

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 4]
```diff
@@ -2503,10 +2540,26 @@ carouselView state push =
...
 homeScreenView :: forall w. (Action -> Effect Unit) -> HomeScreenState -> PrestoDOM (Effect Unit) w
-homeScreenView push state = 
+homeScreenView push state =
+  PrestoAnim.animationSet
+    [ fadeOut (state.props.currentStage == SearchLocationModel)
+    ]
+    $ linearLayout
+        [ height WRAP_CONTENT
+        , width MATCH_PARENT
+        , padding (Padding 0 safeMarginTop 0 safeMarginBottom)
```
Comment (PraveenGongada) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:2553]: PaddingVertical & use `$`

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 5]
```diff
@@ -2960,7 +3013,8 @@ suggestionsView push state =
     , height WRAP_CONTENT
     , orientation VERTICAL
     , padding $ Padding 0 0 0 16
-    , margin $ Margin 8 16 8 0
+    , margin $ Margin 0 16 0 0
```
Comment (PraveenGongada) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:3016]: MarginTop and PaddingBottom

File: Frontend/ui-common/src/Engineering/Helpers/JBridge.purs [hunk 6]
```diff
@@ -254,6 +254,7 @@ foreign import isNetworkTimeEnabled :: EffectFn1 Unit Boolean
 foreign import storeOnResumeCallback :: forall action. Fn2 (action -> Effect Unit) action Unit
 foreign import getLocationNameV2 :: Fn2 Number Number String
 foreign import getLatLonFromAddress :: Fn1 String { latitude :: Number, longitude :: Number }
+foreign import setMapPadding :: EffectFn4 Int Int Int Int Unit
```
Comment (Vignesh-772) [inline Frontend/ui-common/src/Engineering/Helpers/JBridge.purs:257]: Create a wrapper function like below

```
foreign import setMapPaddingImpl :: EffectFn4 Int Int Int Int Unit

setMapPadding :: Int -> Int -> Int -> Int -> Effect Unit
setMapPadding = runEffectFn4 setMapPaddingImpl
```

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs [hunk 7]
```diff
@@ -118,6 +120,11 @@ screen initialState =
             -- push NewUser -- TODO :: Handle the functionality
             _ <- if initialState.data.config.enableMockLocation then isMockLocation push IsMockLocation else pure unit
             _ <- launchAff $ flowRunner defaultGlobalState $ checkForLatLongInSavedLocations push UpdateSavedLoc initialState
+            if(initialState.props.currentStage /= HomeScreen) then do
+              runEffectFn4 setMapPadding 0 0 0 0
```
Comment (Vignesh-772) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:124]: what will happen after ride assigned? route will go behind the bottom sheet ?

Comment (aryan-13) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs:124]: No it won't affect, we are already setting map height to `(V (((screenHeight unit)/ 15)*10))` in `[RideAccepted, RideStarted, ChatWithDriver]` these stages

**Review by witcher-shailesh — APPROVED** [review]

**Pull request merged**